storage/testdb/testdb.go

```diff
@@ -34,14 +35,40 @@ import (
 	_ "github.com/go-sql-driver/mysql" // mysql driver
 )
 
+const (
+	// MySQLURIEnv is the name of the ENV var checked for the test mysql instance URI to use if the dataSourceURI flag is unset.
+	MySQLURIEnv = "MYSQL_URI"
+
+	defaultMySQLURI = "root@tcp(127.0.0.1)/"
+)
+
 var (
 	trillianSQL   = testonly.RelativeToPackage("../mysql/schema/storage.sql")
-	dataSourceURI = flag.String("test_mysql_uri", "root@tcp(127.0.0.1)/", "The MySQL uri to use when running tests")
+	dataSourceURI = flag.String("test_mysql_uri", "", "The MySQL URI to use when running tests. If unset, we'll attempt to use the MYSQL_URI env variable.")
 )
 
+// mysqlURI returns the connection URI to use for tests.
+// It will prefer the value explicitly set by the --test_mysql_uri flag, if
+// that is unset it will fall back to the value in the MYSQL_URI env var, and
+// if that is unset, then it'll try the DefaultMySQLURI value.
+//
+// Only a subset of the suite of tests in this repo require a database it's
+// not possible to blanket apply the test_mysql_uri flag, and maintaining a separate
+// list of tests which needs DB connectivity is brittle, so this fallback approach
+// means we can provide the connection info in a more ambient fashion.
```

Comment:
This really makes me sad and a little frightened. Can we just remove the flag and say that it always uses the env variable to at least remove one layer of magic from this?

It really feels like the root problem is that we're creating multiple different mysql databases, yet wanting to use `testdb` like there is a global static singleton. Trying to solve this is going to be a whole new adventure though.